app/views/samples/_index_bulk_actions_js.haml

```diff
@@ -26,14 +26,12 @@
     });
   }
 
-  function toggleBulkTransfer(checkbox) {
-    var row = checkbox.parents('.laboratory-sample-row')
-    if (checkbox.is(':checked')){
-      if(row.data().isQc){
+  function toggleBulkTransfer() {
+    var areQc = selectedSamplesIds().filter((sample) => sample.isQc === true).length
```

Comment:
You can just ask whether the collection contains a QC sample with [Array.prototype.some()](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/some). This has two benefits: 1. you don't allocate another array and 2. the iteration will stop as soon as a QC sample is found.

```javascript
if (selectedSamplesId().some(sample => sample.isQc)) {
  // at least one selected sample is a QC
}
```